### PR TITLE
[transport] generate log message on crc error

### PR DESCRIPTION
**Pull request by r00t-**
my version of #109

### CODE REVIEW

File: sml/src/sml_message.c [hunk 1]
```diff
@@ -79,10 +79,14 @@ sml_message *sml_message_parse(sml_buffer *buf) {
...
 
-	if (*msg->crc != sml_crc16_calculate(&(buf->buffer[msg_start]), len))
+	if (
+		*msg->crc != sml_crc16_calculate(&(buf->buffer[msg_start]), len) &&
 		// Workaround for Holley DTZ541 uses CRC-16/Kermit
-		if (*msg->crc != sml_crc16kermit_calculate(&(buf->buffer[msg_start]), len))
-			goto error;
+		*msg->crc != sml_crc16kermit_calculate(&(buf->buffer[msg_start]), len)
+	){
+		fprintf(stderr, "libsml: sml_message_parse(): crc mismatch, dropping message\n");
```
Comment (mbehr1) [inline sml/src/sml_message.c:87]: the len of the message might be useful for debugging

**Review by mbehr1 — APPROVED** [review]
lgtm

**Review by jahir — APPROVED** [review]
I agree with @mbehr1, it's often helpful to get some more information in case of errors.

Besides that: lgtm, too

**Comment by r00t-** [comment]
thanks.

i don't think it makes much sense to add much more diagnostics until we have a better way of reporting/logging errors.
(when we should probably give the application the option to retrieve complete raw data for storing/debugging.)

but this is an improvement for now, and i think #103 already covers the topic.

File: sml/src/sml_message.c (hunk 1 above)
Comment (r00t-) [reply sml/src/sml_message.c:87]: a hexdump would be even more useful, but we shouldn't be writing to stderr at all... :\

**Pull request merged**